agent/src/softwarecontaineragent.cpp

```diff
@@ -106,29 +116,35 @@ bool SoftwareContainerAgent::parseConfig(const std::string &config)
 
 ContainerID SoftwareContainerAgent::createContainer(const std::string &prefix, const std::string &config)
 {
-    profilepoint("createContainerStart");
-    profilefunction("createContainerFunction");
-
-    parseConfig(config);
-
-    SoftwareContainer *container;
-    if (m_preloadedContainers.size() != 0) {
-        container = m_preloadedContainers[0].release();
-        m_preloadedContainers.erase(m_preloadedContainers.begin());
-    } else {
-        container = new SoftwareContainer(m_softwarecontainerWorkspace);
-    }
-
-    m_containers.push_back(SoftwareContainerPtr(container));
-    auto id = m_containers.size() - 1;
-    log_debug() << "Created container with ID :" << id;
-    container->setContainerIDPrefix(prefix);
-    container->setMainLoopContext(m_mainLoopContext);
-    container->init();
-
-    triggerPreload();
-
-    return id;
+	profilepoint("createContainerStart");
+	profilefunction("createContainerFunction");
+
+	parseConfig(config);
+
+	SoftwareContainer *container;
+	if (m_preloadedContainers.size() != 0) {
+		container = m_preloadedContainers[0].release();
+		m_preloadedContainers.erase(m_preloadedContainers.begin());
+	} else {
+		container = new SoftwareContainer(m_softwarecontainerWorkspace);
+	}
+
+	ContainerID availableID = m_containerIdPool.back();
```

Comment:
Finding suitable id can be a function you are right. But testing only id pool won't have more meaning testing std::vectors. 
Please see the changes